src/Sentry/Internal/Http/CachingTransport.cs

```diff
@@ -78,6 +78,13 @@ private CachingTransport(ITransport innerTransport, SentryOptions options, bool
             options.TryGetProcessSpecificCacheDirectoryPath() ??
             throw new InvalidOperationException("Cache directory or DSN is not set.");
 
+        // Sanity check: This should never happen in the first place.
+        // We check for `DisableFileWrite` before creating the CachingTransport.
+        if (_options.DisableFileWrite)
```

Comment:
if this never happens consider Debug.Assert. 

Adding `throw` has side effects such as the method can no longer be inlined.